contracts/src/groth16.rs

```diff
@@ -12,74 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use alloy_primitives::U256;
-use alloy_sol_types::{sol, SolValue};
-use anyhow::{ensure, Result};
-use risc0_zkvm::Groth16Seal;
+use alloy_sol_types::SolValue;
+use anyhow::Result;
+use risc0_zkvm::{sha::Digestible, CompactReceipt};
 
-sol! {
-    /// Groth16 seal construction from [RiscZeroGroth16Verifier.sol].
-    ///
-    /// [RiscZeroGroth16Verifier.sol]: https://github.com/risc0/risc0/blob/v0.20.1/bonsai/ethereum/contracts/groth16/RiscZeroGroth16Verifier.sol#L76-L81
-    #[derive(Debug)]
-    struct Seal {
-        uint256[2] a;
-        uint256[2][2] b;
-        uint256[2] c;
-    }
-}
+pub struct Seal {}
```

Comment:
Why have this type? It seems to simply be a container for the ABI encoding methods, and with that I think we might as well drop the struct and simply have the methods live on the module.